src/main/java/org/openrewrite/quarkus/search/FindQuarkusProfiles.java

```diff
@@ -0,0 +1,141 @@
+import org.openrewrite.quarkus.QuarkusExecutionContextView;
+import org.openrewrite.yaml.YamlIsoVisitor;
+import org.openrewrite.yaml.tree.Yaml;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+/**
+ * This recipe is used to find Quarkus properties.
+ */
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class FindQuarkusProfiles extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Search Quarkus profiles";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Search the properties for existing Quarkus profiles.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new TreeVisitor<Tree, ExecutionContext>() {
+                    @Override
+                    public boolean isAcceptable(SourceFile sourceFile, ExecutionContext executionContext) {
+                        QuarkusExecutionContextView quarkusCtx = QuarkusExecutionContextView.view(executionContext);
+                        return sourceFile instanceof Properties.File || sourceFile instanceof Yaml.Documents &&
+                                                                        quarkusCtx.isQuarkusConfigFile(sourceFile.getSourcePath(), null);
+                    }
+                },
+                new FindQuarkusProfilesVisitor()
+        );
+    }
+
+    private static class FindQuarkusProfilesVisitor extends TreeVisitor<Tree, ExecutionContext> {
+
+        @Getter
+        private final Set<String> profiles = new HashSet<>();
+
+        @Override
+        public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+            tree = super.visit(tree, ctx);
+            if (tree instanceof Properties.File) {
+                Properties.File propertiesFile = (Properties.File) tree;
+                new PropertiesIsoVisitor<Set<String>>() {
+                    @Override
+                    public Properties.Entry visitEntry(Properties.Entry entry, Set<String> ctx) {
+                        entry = super.visitEntry(entry, ctx);
+                        addProfile(entry.getKey(), ctx);
+                        return entry;
+                    }
+                }.reduce(propertiesFile, profiles);
+            } else if (tree instanceof Yaml.Documents) {
+                Yaml.Documents yamlDocuments = (Yaml.Documents) tree;
+                new YamlIsoVisitor<Set<String>>() {
+                    @Override
+                    public Yaml.Mapping.Entry visitMappingEntry(Yaml.Mapping.Entry entry, Set<String> ctx) {
+                        entry = super.visitMappingEntry(entry, ctx);
+                        String prop = getProperty(getCursor());
+                        addProfile(prop, ctx);
+                        return entry;
+                    }
+                }.reduce(yamlDocuments, profiles);
+            }
+            return tree;
```

Comment:
I would have expected `SearchResult.found` for these profiles too, as that ensures the results are captured in a data table for use and export, as is customary for our other search recipes. I'm not so sure about the use of a `Set<String> profiles` field for results, as the response from `getVisitor` might be shared, even if the search is limited with preconditions. What are your thoughts there?